src/boosting/sample_strategy.cpp

```diff
@@ -0,0 +1,23 @@
+#include <LightGBM/sample_strategy.h>
+#include "goss.hpp"
+#include "bagging.hpp"
+
+namespace LightGBM {
+
+SampleStrategy* SampleStrategy::CreateSampleStrategy(const Config* config, const Dataset* train_data, const ObjectiveFunction* objective_function, int num_tree_per_iteration) {
```

Comment:
>const Config* config, const Dataset* train_data, const [](http://example.com/codeflow?start=53&length=54)

That is too long, considering expressing them in multiple lines like:

int Func(
     int Param1,
     int Param2,
     int Param3)

Reply:
Addressed via 93a8762